src/intrinsic/simd.rs

```diff
@@ -156,6 +160,163 @@ pub fn generic_simd_intrinsic<'a, 'gcc, 'tcx>(
         return Ok(compare_simd_types(bx, arg1, arg2, in_elem, llret_ty, cmp_op));
     }
 
+    let simd_bswap = |bx: &mut Builder<'a, 'gcc, 'tcx>, vector: RValue<'gcc>| -> RValue<'gcc> {
+        let v_type = vector.get_type();
+        let vector_type = v_type.unqualified().dyncast_vector().expect("vector type");
+        let elem_type = vector_type.get_element_type();
+        let elem_size_bytes = elem_type.get_size();
+        if elem_size_bytes == 1 {
+            return vector;
+        }
+
+        let type_size_bytes = elem_size_bytes as u64 * in_len;
+
+        let shuffle_indices = Vec::from_iter(0..type_size_bytes);
+        let byte_vector_type = bx.context.new_vector_type(bx.type_u8(), type_size_bytes);
+        let byte_vector = bx.context.new_bitcast(None, args[0].immediate(), byte_vector_type);
+
+        let new_elements: Vec<_> = shuffle_indices.chunks_exact(elem_size_bytes as _)
+            .flat_map(|x| x.iter().rev())
+            .map(|&i| {
+                let index = bx.context.new_rvalue_from_long(bx.type_u64(), i as _);
+                bx.extract_element(byte_vector, index)
+            })
+            .collect();
```

Comment:
It might be more efficient to implement this by calling `self.context.new_rvalue_vector_perm()` (see `shuffle_vector`).
What do you think?

Reply:
I'll keep the current implementation around behind a `#[cfg(not(feature = "master"))`, since I'm hoping to keep as much compatibility with non-`master` builds as possible.

See b8bba37 and 0cc4c03.